SplitFiles/bread.c

```diff
@@ -31,12 +32,176 @@ BreadType*  rand_bread_type(BreadBin* bin) {
             }
         }
     }
+    //This should never be reached
+    printf("Error in rand_bread_type");
+    return rand_bread_type(bin);
+}
+
+BreadType* get_bread_type(BreadBin* bin, unsigned char typeLetter) {
```

Comment:
All possible bread types have a letter between A and Z that we turn into an array index  (by subtracting 'A') then use to lookup in bin->breadTypes